src/lib.rs

```diff
@@ -72,7 +72,8 @@ pub use error::AsyncHttpRangeReaderError;
 ///     if response.status() == reqwest::StatusCode::NOT_MODIFIED {
 ///         Ok(None)
 ///     } else {
-///         let reader = AsyncHttpRangeReader::from_head_response(client, response, HeaderMap::default()).await?;
+///         let url = response.url().clone();
+///         let reader = AsyncHttpRangeReader::from_head_response(client, response, url, HeaderMap::default()).await?;
```

Comment:
This, in particular, is awkward. Perhaps we should have a different API instead? Like `AsyncHttpRangeReader::from_head_response_custom_url`?

I feel like similar to #9 it'd be nice to have some sort of `Builder` type that can be constructed and mutated _before_ the `AsyncHttpRangeReader` itself.

Reply:
Yeah I agree that a builder would be a lot nicer now that the options are increasing. Would you be able to wip something up?

Reply:
I can look into it! It might not be quick though :)

Reply:
Sorry if I'm overstepping, but I felt like trying to make a Builder, so I opened #15 😄

Reply:
Oh thanks! I had just asked @charliermarsh to look into this because I was too busy. We'll review it :)